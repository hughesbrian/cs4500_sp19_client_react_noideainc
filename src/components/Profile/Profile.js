import React from 'react'

const Profile = ({firstName, lastName, birthday, address, email, handleChangeInput, updateProfile}) => {
    return(<div className="container">
        <h1>Profile</h1>
        <a className="btn btn-light" href="/business" role="button">My Business</a>
        <span> </span>
        <a className="btn btn-light" href="/services" role="button">My Services</a>
        <br/>
        <br/>
        <form onSubmit={updateProfile}>
            <h4>Legal name</h4>
            <div className="row">
                <div className="col-6">
                    <label htmlFor="first-name">First name</label>
                    <input id="first-name" 
                            name="firstName"
                            className="form-control" 
                            value={firstName}
                            onChange={handleChangeInput}/>
                </div>
                <div className="col-6">
                    <label htmlFor="last-name">Last name</label>
                    <input id="last-name" 
                            name="lastName"
                            className="form-control"
                            value={lastName}
                            onChange={handleChangeInput}/>
                </div>
            </div>
            <br/>
            <h4>Date of birth</h4>
            <div className="row">
                <div className="col-4">
                    <input className="form-control" 
                            type="date" 
                            name="birthday" 
                            value={birthday}
                            onChange={handleChangeInput}></input>
                </div>
            </div>
            <br/>
            <h4>Home address</h4>
            <div className="row">
                <div className="col-12">
                    <label htmlFor="street">Street</label>
                    <input id="street" 
                            type="text"
                            name="street"
                            className="form-control"
                            value={address.street}
                            onChange={handleChangeInput}
                            />
                </div>
            </div>
            <div className="row">
                <div className="col-12">
                    <label htmlFor="city">City</label>
                    <input id="city" name="city" className="form-control" 
                    value={address.city} 
                    onChange={handleChangeInput}/>
                </div>
            </div>
            <div className="row">
                <div className="col-6">
                    <label htmlFor="state">State</label>
                    <select name="state" className="form-control" value={address.state} onChange={handleChangeInput}>
                        <option value="MA">MA</option>
                        <option value="NH">NH</option>
                        <option value="NY">NY</option>
                        <option value="CA">CA</option>
                    </select>
                </div>
                <div className="col-6">
                    <label htmlFor="zip">Zip</label>
                    <input id="zip" name="zip" className="form-control" 
                    value={address.zip} 
                    onChange={handleChangeInput}/>
                </div>
            </div>
            <br/>
            <div className="row">
                <div className="col-12">
                    <label htmlFor="email">Email</label>
                    <input id="email" 
                            name="email"
                            className="form-control"
                            value={email}
                            readOnly={true}/>
                </div>
            </div>
            <br/>
            <div className="row">
                <div className="col-12">
                    <input className="btn btn-success btn-block" type="submit" value="Update Account"></input>
                </div>
            </div>
            <br/>
            <br/>
            <br/>
            <br/>
        </form>
    </div>)
}

export default Profile