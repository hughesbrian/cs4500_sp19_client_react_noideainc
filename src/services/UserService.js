export default class UserService {
    static instance = null;

    static getInstance() {
        if (UserService.instance === null) {
            UserService.instance = new UserService()
        }
        return this.instance
    }

    findUserById = userId =>
        fetch("https://cs4500-sp19-noideainc.herokuapp.com/api/users/${userId}")
            .then(response => response.json())
    findAllUsers = () =>
        fetch("https://cs4500-sp19-noideainc.herokuapp.com/api/users")
            .then(response => response.json())
}