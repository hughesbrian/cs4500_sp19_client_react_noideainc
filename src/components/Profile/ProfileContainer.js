import React from 'react'
import Profile from '../../components/Profile/Profile'
import userService from '../../services/UserService'

class ProfileContainer extends React.Component {
    constructor(props) {
        super(props);
        this.userService = userService.getInstance();
        this.state = {
            user: {},
            userId: props.userId,
            firstName: props.user.firstName,
            lastName: props.user.lastName,
            birthday: props.user.birthday,
            address: {
                street: "",
                city: "",
                state: "",
                zip: "",
                addressType: 0,
                residentId: props.userId
            },
            email: ""
        }
        this.businessAddress = null
        this.handleChangeInput = this.handleChangeInput.bind(this)
        this.updateProfile = this.updateProfile.bind(this)
    }

    componentDidMount() {
        this.userService
            .findUserById(this.state.userId)
            .then(user => {
                this.setState({
                    user: user,
                    firstName: user.firstName,
                    lastName: user.lastName,
                    birthday: user.birthday,
                    email: user.email,
                    address: user.addresses[0]
                })
            })
    }

    handleChangeInput(event) {
        event.preventDefault();
        this.setState({
            ...this.state,
            address: {
                ...this.state.address,
                [event.target.name]: event.target.value
            },
            [event.target.name]: event.target.value
        });
    }

    updateProfile(event) {
        event.preventDefault();
        // console.log("form submitted")
        const newData = {
            id: this.state.userId,
            firstName: this.state.firstName,
            lastName: this.state.lastName,
            birthday: this.state.birthday,
            addresses: this.state.user.addresses,
            email: this.state.email
        }
        newData.addresses[0].id = this.state.address.id
        newData.addresses[0].street = this.state.address.street
        newData.addresses[0].city = this.state.address.city
        newData.addresses[0].state = this.state.address.state
        newData.addresses[0].zip = this.state.address.zip

        this.userService
            .updateProfile(newData)
            .then(alert("profile updated sucessfully"))

    }

    render = () => {
        return <Profile
                    userId={this.state.userId}
                    firstName={this.state.firstName}
                    lastName={this.state.lastName}
                    birthday={this.state.birthday}
                    address={this.state.address}
                    email={this.state.email}
                    handleChangeInput={this.handleChangeInput}
                    updateProfile={this.updateProfile}
                />
    }
}

export default ProfileContainer;
