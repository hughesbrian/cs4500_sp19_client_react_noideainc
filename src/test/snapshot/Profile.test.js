import React from 'react'
import ProfileContainer from '../../components/Profile/ProfileContainer'
import Profile from '../../components/Profile/Profile'
import TestRenderer from 'react-test-renderer';
import UserService from '../../services/UserService'
import userMock from '../MockData/UserProfile.mock.json';

const userService = UserService.getInstance();

test('User profile renders correctly', () => {
    const testRenderer = TestRenderer.create(
        <ProfileContainer
            userService={userService}
            user={userMock}
            />
        )
            let tree = testRenderer.toJSON();
            expect(tree).toMatchSnapshot()
});

test('User profile updates correctly', () => {
    // dummy functions
    const updateProfile = jest.fn();
    const handleChangeInput = jest.fn();

    const testRenderer = TestRenderer.create(
        <Profile
            userId={userMock.id}
            firstName={userMock.firstName}
            lastName={userMock.lastName}
            birthday={userMock.birthday}
            address={userMock.address}
            email={userMock.email}
            handleChangeInput={handleChangeInput}
            updateProfile={updateProfile}
        />
    )

    let tree = testRenderer.toJSON();
    const testInstance = testRenderer.root
    const profileForm =  testInstance.findByProps({ id: 'prifle-form' });
    const firstNameField = testInstance.findByProps({ id: 'first-name' })
    const lastNameField = testInstance.findByProps({ id: 'last-name' })
    const birthdayField = testInstance.findByProps({ id: 'user-birthday' })
    const streetField = testInstance.findByProps({ id: 'street' })
    const cityField = testInstance.findByProps({ id: 'city' })
    const stateField = testInstance.findByProps({ id: 'user-state' })
    const zipField = testInstance.findByProps({ id: 'zip' })
    const inputFields = testInstance.findAllByProps({ className: 'form-control' })

    birthdayField.props.onChange()
    streetField.props.onChange()
    cityField.props.onChange()
    stateField.props.onChange()
    zipField.props.onChange()
    profileForm.props.onSubmit()

    expect(handleChangeInput).toHaveBeenCalled();
    expect(updateProfile).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
    expect(inputFields.length).toBe(8)

    // check input values
    expect(firstNameField.props.value).toBe("Bob")
    expect(lastNameField.props.value).toBe("Wonder")
    expect(birthdayField.props.value).toBe("1/01/1994")
    expect(streetField.props.value).toBe("123 Washington St.")
    expect(cityField.props.value).toBe("Boston")
    expect(stateField.props.value).toBe("MA")
    expect(zipField.props.value).toBe("02117")
})
