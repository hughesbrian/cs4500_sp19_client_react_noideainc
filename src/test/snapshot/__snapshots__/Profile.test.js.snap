// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User profile renders correctly 1`] = `
<div
  className="container"
>
  <h1>
    Profile
  </h1>
  <a
    className="btn btn-light"
    href="/Profile/undefined/business/"
    role="button"
  >
    My Business
  </a>
  <span>
     
  </span>
  <a
    className="btn btn-light"
    href="/services"
    role="button"
  >
    My Services
  </a>
  <br />
  <br />
  <form
    id="prifle-form"
    onSubmit={[Function]}
  >
    <h4>
      Legal name
    </h4>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          htmlFor="first-name"
        >
          First name
        </label>
        <input
          className="form-control"
          id="first-name"
          name="firstName"
          onChange={[Function]}
          value="Bob"
        />
      </div>
      <div
        className="col-6"
      >
        <label
          htmlFor="last-name"
        >
          Last name
        </label>
        <input
          className="form-control"
          id="last-name"
          name="lastName"
          onChange={[Function]}
          value="Wonder"
        />
      </div>
    </div>
    <br />
    <h4>
      Date of birth
    </h4>
    <div
      className="row"
    >
      <div
        className="col-4"
      >
        <input
          className="form-control"
          id="user-birthday"
          name="birthday"
          onChange={[Function]}
          type="date"
          value="1/01/1994"
        />
      </div>
    </div>
    <br />
    <h4>
      Home address
    </h4>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          htmlFor="street"
        >
          Street
        </label>
        <input
          className="form-control"
          id="street"
          name="street"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          htmlFor="city"
        >
          City
        </label>
        <input
          className="form-control"
          id="city"
          name="city"
          onChange={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          htmlFor="state"
        >
          State
        </label>
        <select
          className="form-control"
          id="user-state"
          name="state"
          onChange={[Function]}
          value=""
        >
          <option
            value="AL"
          >
            AL
          </option>
          <option
            value="AK"
          >
            AK
          </option>
          <option
            value="AZ"
          >
            AZ
          </option>
          <option
            value="AR"
          >
            AR
          </option>
          <option
            value="CA"
          >
            CA
          </option>
          <option
            value="CO"
          >
            CO
          </option>
          <option
            value="CT"
          >
            CT
          </option>
          <option
            value="DE"
          >
            DE
          </option>
          <option
            value="FL"
          >
            FL
          </option>
          <option
            value="GA"
          >
            GA
          </option>
          <option
            value="HI"
          >
            HI
          </option>
          <option
            value="ID"
          >
            ID
          </option>
          <option
            value="IL"
          >
            IL
          </option>
          <option
            value="IN"
          >
            IN
          </option>
          <option
            value="IA"
          >
            IA
          </option>
          <option
            value="KS"
          >
            KS
          </option>
          <option
            value="KY"
          >
            KY
          </option>
          <option
            value="LA"
          >
            LA
          </option>
          <option
            value="ME"
          >
            ME
          </option>
          <option
            value="MD"
          >
            MD
          </option>
          <option
            value="MA"
          >
            MA
          </option>
          <option
            value="MI"
          >
            MI
          </option>
          <option
            value="MN"
          >
            MN
          </option>
          <option
            value="MS"
          >
            MS
          </option>
          <option
            value="MO"
          >
            MO
          </option>
          <option
            value="MT"
          >
            MT
          </option>
          <option
            value="NE"
          >
            NE
          </option>
          <option
            value="NV"
          >
            NV
          </option>
          <option
            value="NH"
          >
            NH
          </option>
          <option
            value="NJ"
          >
            NJ
          </option>
          <option
            value="NM"
          >
            NM
          </option>
          <option
            value="NY"
          >
            NY
          </option>
          <option
            value="NC"
          >
            NC
          </option>
          <option
            value="ND"
          >
            ND
          </option>
          <option
            value="OH"
          >
            OH
          </option>
          <option
            value="OK"
          >
            OK
          </option>
          <option
            value="OR"
          >
            OR
          </option>
          <option
            value="PA"
          >
            PA
          </option>
          <option
            value="RI"
          >
            RI
          </option>
          <option
            value="SC"
          >
            SC
          </option>
          <option
            value="SD"
          >
            SD
          </option>
          <option
            value="TN"
          >
            TN
          </option>
          <option
            value="TX"
          >
            TX
          </option>
          <option
            value="UT"
          >
            UT
          </option>
          <option
            value="VT"
          >
            VT
          </option>
          <option
            value="VA"
          >
            VA
          </option>
          <option
            value="WA"
          >
            WA
          </option>
          <option
            value="WV"
          >
            WV
          </option>
          <option
            value="WI"
          >
            WI
          </option>
          <option
            value="WY"
          >
            WY
          </option>
        </select>
      </div>
      <div
        className="col-6"
      >
        <label
          htmlFor="zip"
        >
          Zip
        </label>
        <input
          className="form-control"
          id="zip"
          name="zip"
          onChange={[Function]}
          value=""
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          readOnly={true}
          value=""
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <input
          className="btn btn-success btn-block"
          id="submit-profile"
          type="submit"
          value="Update Account"
        />
      </div>
    </div>
    <br />
    <br />
    <br />
    <br />
  </form>
</div>
`;

exports[`User profile updates correctly 1`] = `
<div
  className="container"
>
  <h1>
    Profile
  </h1>
  <a
    className="btn btn-light"
    href="/Profile/555/business/"
    role="button"
  >
    My Business
  </a>
  <span>
     
  </span>
  <a
    className="btn btn-light"
    href="/services"
    role="button"
  >
    My Services
  </a>
  <br />
  <br />
  <form
    id="prifle-form"
    onSubmit={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  >
    <h4>
      Legal name
    </h4>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          htmlFor="first-name"
        >
          First name
        </label>
        <input
          className="form-control"
          id="first-name"
          name="firstName"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
                Array [],
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          value="Bob"
        />
      </div>
      <div
        className="col-6"
      >
        <label
          htmlFor="last-name"
        >
          Last name
        </label>
        <input
          className="form-control"
          id="last-name"
          name="lastName"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
                Array [],
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          value="Wonder"
        />
      </div>
    </div>
    <br />
    <h4>
      Date of birth
    </h4>
    <div
      className="row"
    >
      <div
        className="col-4"
      >
        <input
          className="form-control"
          id="user-birthday"
          name="birthday"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
                Array [],
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          type="date"
          value="1/01/1994"
        />
      </div>
    </div>
    <br />
    <h4>
      Home address
    </h4>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          htmlFor="street"
        >
          Street
        </label>
        <input
          className="form-control"
          id="street"
          name="street"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
                Array [],
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          type="text"
          value="123 Washington St."
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          htmlFor="city"
        >
          City
        </label>
        <input
          className="form-control"
          id="city"
          name="city"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
                Array [],
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          value="Boston"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          htmlFor="state"
        >
          State
        </label>
        <select
          className="form-control"
          id="user-state"
          name="state"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
                Array [],
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          value="MA"
        >
          <option
            value="AL"
          >
            AL
          </option>
          <option
            value="AK"
          >
            AK
          </option>
          <option
            value="AZ"
          >
            AZ
          </option>
          <option
            value="AR"
          >
            AR
          </option>
          <option
            value="CA"
          >
            CA
          </option>
          <option
            value="CO"
          >
            CO
          </option>
          <option
            value="CT"
          >
            CT
          </option>
          <option
            value="DE"
          >
            DE
          </option>
          <option
            value="FL"
          >
            FL
          </option>
          <option
            value="GA"
          >
            GA
          </option>
          <option
            value="HI"
          >
            HI
          </option>
          <option
            value="ID"
          >
            ID
          </option>
          <option
            value="IL"
          >
            IL
          </option>
          <option
            value="IN"
          >
            IN
          </option>
          <option
            value="IA"
          >
            IA
          </option>
          <option
            value="KS"
          >
            KS
          </option>
          <option
            value="KY"
          >
            KY
          </option>
          <option
            value="LA"
          >
            LA
          </option>
          <option
            value="ME"
          >
            ME
          </option>
          <option
            value="MD"
          >
            MD
          </option>
          <option
            value="MA"
          >
            MA
          </option>
          <option
            value="MI"
          >
            MI
          </option>
          <option
            value="MN"
          >
            MN
          </option>
          <option
            value="MS"
          >
            MS
          </option>
          <option
            value="MO"
          >
            MO
          </option>
          <option
            value="MT"
          >
            MT
          </option>
          <option
            value="NE"
          >
            NE
          </option>
          <option
            value="NV"
          >
            NV
          </option>
          <option
            value="NH"
          >
            NH
          </option>
          <option
            value="NJ"
          >
            NJ
          </option>
          <option
            value="NM"
          >
            NM
          </option>
          <option
            value="NY"
          >
            NY
          </option>
          <option
            value="NC"
          >
            NC
          </option>
          <option
            value="ND"
          >
            ND
          </option>
          <option
            value="OH"
          >
            OH
          </option>
          <option
            value="OK"
          >
            OK
          </option>
          <option
            value="OR"
          >
            OR
          </option>
          <option
            value="PA"
          >
            PA
          </option>
          <option
            value="RI"
          >
            RI
          </option>
          <option
            value="SC"
          >
            SC
          </option>
          <option
            value="SD"
          >
            SD
          </option>
          <option
            value="TN"
          >
            TN
          </option>
          <option
            value="TX"
          >
            TX
          </option>
          <option
            value="UT"
          >
            UT
          </option>
          <option
            value="VT"
          >
            VT
          </option>
          <option
            value="VA"
          >
            VA
          </option>
          <option
            value="WA"
          >
            WA
          </option>
          <option
            value="WV"
          >
            WV
          </option>
          <option
            value="WI"
          >
            WI
          </option>
          <option
            value="WY"
          >
            WY
          </option>
        </select>
      </div>
      <div
        className="col-6"
      >
        <label
          htmlFor="zip"
        >
          Zip
        </label>
        <input
          className="form-control"
          id="zip"
          name="zip"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
                Array [],
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          value="02117"
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          readOnly={true}
          value="bob@gmail.com"
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <input
          className="btn btn-success btn-block"
          id="submit-profile"
          type="submit"
          value="Update Account"
        />
      </div>
    </div>
    <br />
    <br />
    <br />
    <br />
  </form>
</div>
`;
